sdk/keyvault/azure-security-keyvault-certificates/src/certificate_serializers.cpp

```diff
@@ -489,3 +489,47 @@ std::string CertificateIssuerSerializer::Serialize(CertificateIssuer const& issu
 
   return jsonResponse.dump();
 }
+
+std::string CertificateContactsSerializer::Serialize(
+    std::vector<CertificateContact> const& contacts)
+{
+  json payload;
+
+  for (auto contact : contacts)
+  {
+    json contactJson;
+
+    contactJson[EmailPropertyName] = contact.EmailAddress;
+    JsonOptional::SetFromNullable(contact.Name, contactJson, NamePropertyName);
+    JsonOptional::SetFromNullable(contact.Phone, contactJson, PhonePropertyName);
+
+    payload[ContactsPropertyName].emplace_back(contactJson);
+  }
+
+  return payload.dump();
+}
+
+std::vector<CertificateContact> CertificateContactsSerializer::Deserialize(
+    Azure::Core::Http::RawResponse const& rawResponse)
+{
+  std::vector<CertificateContact> response;
+
+  auto const& body = rawResponse.GetBody();
+  auto jsonResponse = json::parse(body);
+
+  if (jsonResponse.contains(ContactsPropertyName))
+  {
+    for (auto contactJson : jsonResponse[ContactsPropertyName])
+    {
+      CertificateContact contact;
+
+      contact.EmailAddress = contactJson[EmailPropertyName];
+      JsonOptional::SetIfExists(contact.Name, contactJson, NamePropertyName);
+      JsonOptional::SetIfExists(contact.Phone, contactJson, PhonePropertyName);
+
+      response.emplace_back(contact);
+    }
+  }
+
+  return response;
```

Comment:
It probably is fine, but just to confirm, is it the right behavior to just return an empty/uninitialized vector if the response json doesn't contain the contacts property?

Reply:
yes, for one thing and std::vector can be empty but not null, 
second
the .Net deserialization does the same thing sets the _contacts property to empty list even is the node is missing in json 
 void IJsonDeserializable.ReadProperties(JsonElement json)
        {
            var contacts = new List<CertificateContact>();

            if (json.TryGetProperty(ContactsPropertyName, out JsonElement contactsElement))
            {
                foreach (JsonElement entry in contactsElement.EnumerateArray())
                {
                    var contact = new CertificateContact();

                    ((IJsonDeserializable)contact).ReadProperties(entry);

                    contacts.Add(contact);
                }
            }

            _contacts = contacts;
        }